### DISCUSSION TITLE
Foundry Fundamentals: Section 1 Lesson 23. Failed to deploy on sepolia testnet

### DISCUSSION THREAD

**Discussion by UmesiQueen** [Cyfrin/foundry-full-course-cu#2846] [Q&A] [answered]
I assume this lesson has not been updated to use the keystore method, so I went ahead to try it out myself but ran into an error. 

Firstly, I successfully imported my wallet using
```bash 
cast wallet import --account-name metamask1 --interactive
```

and I deployed my script with the command, 
```bash
forge script script/DeploySimpleStorage.s.sol --rpc-url $ZKSYNC_RPC_URL --account  metamask1 --sender 0x419AE9a...dA9Ad
```

It compiled, script ran successfully and It requested for my keystore password too. Meaning I should have imported my wallet correctly, I guess. However, at the end it failed to deploy. This is the error message I got.

> Error: Failed to send transaction
> 
> Context:
> - server returned an error response: error code 3: Failed to serialize transaction: toAddressIsNull

I do not know why this happened.

**Comment by heheboii11** [comment]
As far as I can see, you are trying to send transaction to a null address, i.e, the address you want to send might not be in the array. If you can provide the ```deployscript``` and ```simpleStorage``` ```src```file, it would be easy to point out exact error

**Comment by EngrPips** [comment, marked as answer]
Hello @UmesiQueen, Foundry Script doesn't work with `ZKSync` at the moment. Please deploy to sepolia instead

**Reply by UmesiQueen** [reply]
Thank you @EngrPips. It works

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.
